jazzhands/__main__.py

```diff
@@ -184,25 +209,57 @@ def main(argv=sys.argv):
     # collect, build, run or the setup command.
     # build implies collect, run implies build and collect. setup runs on its own.
 
-    if len(sys.argv) <= 1:
+    parser = argparse.ArgumentParser()
+    parser.set_defaults(which=None)
+    # parser.add_argument('command', type=str, action='store')
+    subparsers = parser.add_subparsers()
+    parsers = {}
+    def add_command(cmd):
+        parsers[cmd] = subparsers.add_parser(cmd)
+        parsers[cmd].set_defaults(which=cmd)
+
+    add_command('setup')
+    parsers['setup'].add_argument('-p', '--preset', action='append', type=str, default=[])
+    parsers['setup'].add_argument('-t', '--transform', action='append', type=str, default=[])
+
+    add_command('collect')
+
+    add_command('build')
+
+    add_command('run')
+    parsers['run'].add_argument('--auto-npm', action='store_true')
+
+    args = parser.parse_args()
+
+    if not args.which:
         DO_COLLECT = DO_BUILD = True
-    elif sys.argv[1] == 'setup':
+    elif args.which == 'setup':
         # Do one time setup stuff for the project
-        args = [
+        # Currently defaults to Babel's ES2017
+        cargs = [
             "npm", "install", "--save",
             "babel-core",
             "babel-cli",
             "babelify",
-            "babel-preset-react",
-            "babel-preset-es2015",
         ]
-        subprocess.call(args)
+        for preset in args.preset:
+            cargs.append("babel-preset-" + preset)
+        for transform in args.transform:
+            cargs.append("babel-plugin-transform-" + transform)
+
+        babelrc = {
```

Comment:
Should we check whether a .babelrc file already exists and not overwrite it?  Maybe just print out what we would have done and prompt the user to update their .babelrc manually if necessary?